tool/generate_new_sample.dart

```diff
@@ -0,0 +1,169 @@
+      sampleDirectory.path,
+    );
+  }
+
+  // Create the sample directory
+  try {
+    sampleDirectory.createSync();
+    print('>Sample directory created at ${sampleDirectory.path}');
+  } on Exception catch (_) {
+    rethrow;
+  }
+
+  // Create the README.md file
+  createEmptyReadMeOrCopy(sampleDirectory, sampleCamelName);
+
+  // Create the sample file
+  createNewSampleFile(sampleDirectory, sampleSnakeName, sampleCamelName);
+
+  // Add the sample to the samples_widget_list.dart file
+  addSampleToSamplesWidgetList(
+      sampleRootDirectory, sampleSnakeName, sampleCamelName);
+}
+
+// Convert a camel case string to snake case.
+String camelToSnake(String input) {
+  final snakeCase = input.replaceAllMapped(
+    RegExp(r'([a-z])([A-Z])'),
+    (Match match) => '${match.group(1)}_${match.group(2)!.toLowerCase()}',
+  );
+  return snakeCase.toLowerCase();
+}
+
+// Create a new sample README.md file,
+// or copy the template README.md file
+// from the common-samples/designs directory if it exists.
+// The common-samples directory is expected to be at the same level
+// as the samples directory.
+// - /common-samples
+// - /arcgis-maps-sdk-flutter-samples
+void createEmptyReadMeOrCopy(
+    Directory sampleDirectory, String sampleCamelName) {
+  final ps = Platform.pathSeparator;
+  final templateReadmeFile = File(
+      '${Directory.current.parent.path}${ps}common-samples${ps}designs$ps$sampleCamelName${ps}README.md');
+  try {
+    final sampleReadmeFile = File('${sampleDirectory.path}${ps}README.md');
+    if (templateReadmeFile.existsSync()) {
+      sampleReadmeFile.writeAsBytesSync(templateReadmeFile.readAsBytesSync());
+      print('>A README File created');
+    } else {
+      print('>A empty README file was created');
+      sampleReadmeFile.writeAsStringSync('README-Empty');
+    }
+  } catch (_) {
+    rethrow;
+  }
+}
+
+// Create a new sample file
+void createNewSampleFile(
+    Directory sampleDirectory, String sampleSnakeName, String sampleCamelName) {
+  final ps = Platform.pathSeparator;
+  final templateFile = File(
+      '${Directory.current.parent.path}${ps}flutter${ps}internal${ps}lib${ps}sample_skeleton.dart');
+  final sampleFile = File('${sampleDirectory.path}$ps$sampleSnakeName.dart');
+
+  sampleFile.createSync();
+  sampleFile.writeAsStringSync(copyright);
+
+  if (templateFile.existsSync()) {
+    final lines = templateFile.readAsLinesSync();
+    var skip = false;
+    for (var line in lines) {
+      skip = line.startsWith('//') ? true : false;
+      if (!skip) {
+        final newLine = line.replaceAll('SampleWidget', sampleCamelName);
+        sampleFile.writeAsStringSync('$newLine${Platform.lineTerminator}',
+            mode: FileMode.append);
+      }
+    }
+  }
+  print('>A sample file created');
+}
+
+// Add the new sample to the samples_widget_list.dart file
+void addSampleToSamplesWidgetList(Directory sampleRootDirectory,
+    String sampleSnakeName, String sampleCamelName) {
```

Comment:
The import lines won't be in alphabetical order, which will cause an analyzer warning. I think a better approach would be to write the whole file from scratch. That way it doesn't have to worry about inserting the new lines, it can just write out everything in the correct order.

1. Read the directory `lib/samples/` using `Directory.listSync()` and get the names of all subdirectories
2. Sort the list of subdirectories alphabetically
3. For each subdirectory, print out the `import` statement
4. Print out the declaration and opening bracket for `final sampleWidgets`
5. For each subdirectory, print out the `sampleWidgets` entry (e.g., `'display_map': () => const DisplayMap(),`)
6. Print out the closing bracket
7. run `dart format` on the file